app/src/main/java/com/android/unio/ui/saved/Saved.kt

```diff
@@ -1,28 +1,128 @@
 package com.android.unio.ui.saved
 
-import android.annotation.SuppressLint
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Place
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
 import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
 import com.android.unio.R
+import com.android.unio.model.event.EventViewModel
 import com.android.unio.model.strings.test_tags.SavedTestTags
+import com.android.unio.model.user.UserViewModel
+import com.android.unio.ui.event.EventCard
 import com.android.unio.ui.navigation.BottomNavigationMenu
 import com.android.unio.ui.navigation.LIST_TOP_LEVEL_DESTINATION
 import com.android.unio.ui.navigation.NavigationAction
 import com.android.unio.ui.navigation.Route
+import com.android.unio.ui.navigation.Screen
+import com.android.unio.ui.theme.AppTypography
+import java.util.Calendar
 
-@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
 @Composable
-fun SavedScreen(navigationAction: NavigationAction) {
+fun SavedScreen(
+    navigationAction: NavigationAction,
+    eventViewModel: EventViewModel,
+    userViewModel: UserViewModel
+) {
+  val allEvents by eventViewModel.events.collectAsState()
+  val savedEvents = allEvents.filter { userViewModel.isEventSavedForCurrentUser(it.uid) }
   val context = LocalContext.current
+
+  val today = remember { Calendar.getInstance().get(Calendar.DAY_OF_YEAR) }
+  val savedEventsToday = remember {
+    savedEvents.partition {
+      val calendar = Calendar.getInstance()
+      calendar.timeInMillis = it.startDate.toDate().time
+      calendar.get(Calendar.DAY_OF_YEAR) == today
+    }
```

Comment:
In my opinion it would be better to divide this pair into 2 variables : savedEventsToday and savedEventsUpcoming so the name is less confusing. The UI separation between today and upcoming is a good idea though and it is nicely coded !

Reply:
And I agree that the name of savedEventsToday  could be changed to something like savedEventsPartitioned or at least something that better reflects its nature, but I like the solutin with the .partition and find it clean. It avoids code duplication and is more efficient

Reply:
You're right the partition is nice, I was only talking of then dividing the pair into 2 different lists